Tests/Integration/JoinSlowTwinUpdateTests.cs

```diff
@@ -41,31 +41,29 @@ public async Task When_First_Join_Fails_Due_To_Slow_Twin_Update_Retry_Second_Att
                 .ReturnsAsync(twin);
 
             // Device twin will be updated
-            AppSessionKey? afterJoin1AppSKey = null;
-            NetworkSessionKey? afterJoin1NwkSKey = null;
-            string afterJoin1DevAddr = null;
             AppSessionKey? afterJoin2AppSKey = null;
             NetworkSessionKey? afterJoin2NwkSKey = null;
             string afterJoin2DevAddr = null;
 
-            LoRaDeviceClient.Setup(x => x.UpdateReportedPropertiesAsync(It.IsNotNull<TwinCollection>()))
-                .ReturnsAsync(true, TimeSpan.FromSeconds(10))
-                .Callback<TwinCollection>((updatedTwin) =>
-                {
-                    afterJoin1AppSKey = AppSessionKey.Parse(updatedTwin[TwinProperty.AppSKey].Value);
-                    afterJoin1NwkSKey = NetworkSessionKey.Parse(updatedTwin[TwinProperty.NwkSKey].Value);
-                    afterJoin1DevAddr = updatedTwin[TwinProperty.DevAddr];
-
-                    // update setup for no delay
-                    LoRaDeviceClient.Setup(x => x.UpdateReportedPropertiesAsync(It.IsNotNull<TwinCollection>()))
-                        .ReturnsAsync(true)
-                        .Callback<TwinCollection>((updatedTwin2) =>
-                        {
-                            afterJoin2AppSKey = AppSessionKey.Parse(updatedTwin2[TwinProperty.AppSKey].Value);
-                            afterJoin2NwkSKey = NetworkSessionKey.Parse(updatedTwin2[TwinProperty.NwkSKey].Value);
-                            afterJoin2DevAddr = updatedTwin2[TwinProperty.DevAddr];
-                        });
-                });
+            var mockSequence = new MockSequence();
+            LoRaDeviceClient.InSequence(mockSequence)
+                            .Setup(x => x.UpdateReportedPropertiesAsync(It.IsNotNull<TwinCollection>(), It.IsAny<CancellationToken>()))
+                            .Returns<TwinCollection, CancellationToken>(async (_, token) =>
+                            {
+                                // No cancellation token specified because we want to wait 7 seconds no matter what, to make the "token" being cancelled by logic
+                                await Task.Delay(7_000, CancellationToken.None);
+                                Assert.True(token.IsCancellationRequested, "First request should have been canceled after timeout.");
+                                return false;
```

Comment:
The 7s are hard to understand for anyone not knowing the 2n downtime window is 6s. I would use the token - that also makes the test more resilient for future changes to that that window. Something like that?
```suggestion
                                // No cancellation token specified because we want to wait 7 seconds no matter what, to make the "token" being cancelled by logic
                                try
                                {
                                   await Task.Delay(TimeSpan.FromSeconds(20), token);
                                   Assert.True(false, "Token timeout expected");
                                }
                                catch (OperationCanceledException) { }
                                return false;
```

Reply:
Done in 86963a1